geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/CacheClientNotifier.java

```diff
@@ -608,7 +608,7 @@ public static boolean singletonHasClientProxies() {
   }
 
   private boolean hasClientProxies() {
-    return !this._initClientProxies.isEmpty() || !this._clientProxies.isEmpty();
+    return !this._clientProxies.isEmpty() || !this._initClientProxies.isEmpty();
```

Comment:
0. initial condition: _clientProxies.isEmpty() and !_initClientProxies.isEmpty()
1. Thread 1 calls _clientProxies.isEmpty() which returns true
2. Thread 2 calls addClientProxy so now: !_clientProxies.isEmpty() and _initClientProxies.isEmpty()
3. Thread 1 calls _initClientProxies.isEmpty() which returns true
So Thread 1 returns false from hasClientProxies even though we have one client proxy.
The fix is to check _initClientProxies first and then _clientProxies

Reply:
Sorry, but when resolving merge conflicts, I deleted wrong part.